schemas/errors/generator.py

```diff
@@ -0,0 +1,98 @@
+import json
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List
+
+
+@dataclass
+class ErrorDefinition:
+    code: str
+    http_status_code: int
+    description: str
+
+
+class ErrorGenerator(ABC):
+    @abstractmethod
+    def generate(self, errors: List[ErrorDefinition]) -> str:
+        """Generate error code implementation from error definitions"""
+        pass
+
+    @abstractmethod
+    def get_output_directory(self) -> str:
+        """Get the output directory"""
+        pass
+
+    @abstractmethod
+    def get_output_file_name(self) -> str:
+        """Get the output file name"""
+        pass
+
+
+class PythonErrorGenerator(ErrorGenerator):
```

Comment:
Might make sense to break this out to another file since we'll need one for each language.